src/elli/importer/nexus.py

```diff
@@ -30,29 +54,86 @@ def read_nexus_psi_delta(nxs_filename: str) -> pd.DataFrame:
         The index is a multiindex consisting of the angle of incidents as first column
         and the wavelength as second column.
     """
+
+    def read_data(
+        wavelength_path: str,
+        aois_path: str,
+        data_path: str,
+        indexing: Callable[[int, int], IndexExpression],
+    ):
+        aois = np.array(h5file[aois_path])
+        wavelength = np.array(h5file[wavelength_path]) / 10
+        psi_delta_df = pd.DataFrame(
+            {},
+            columns=["Ψ", "Δ"],
+            index=pd.MultiIndex.from_product(
+                [aois, wavelength], names=["Angle of Incidence", "Wavelength"]
+            ),
+            dtype=float,
+        )
+
+        data = np.array(h5file[data_path])
+
+        for i, aoi in enumerate(aois):
+            psi_delta_df.loc[aoi, "Ψ"] = data[indexing(i, 0)]
+            psi_delta_df.loc[aoi, "Δ"] = data[indexing(i, 1)]
```

Comment:
So we use this as default data structure for now, which i am totally fine with.

But i am not sure if it breaks calc_rho and fitting for multi aio files. But i have no way to test that quickly.

Reply:
yeah calc_rho essentially assumes this structure and so its what were using rn. Anyways, I think we should change this to a general proper data structure eventually but then also w/ support for arbitrary scan axis.